### PR TITLE
Fix issue with fields trying to use wrong name

### PR DESCRIPTION
**Pull request by PMunch**
While trying to implement a `hash` procedure for `Sockaddr_storage` in the posix module I got some issues like this:

```
/home/peter/.cache/nim/client_d/stdlib_dollars.nim.c: In function ‘dollar___5tyEFhTc0GQEDhveAS9aQpw’:
/home/peter/.cache/nim/client_d/stdlib_dollars.nim.c:244:49: error: ‘struct sockaddr_storage’ has no member named ‘ss_padding’; did you mean ‘__ss_padding’?
  244 |  addQuoted__Z2JuWoDGzsNYTOa0dt2Kgw((&result), x.ss_padding);
      |                                                 ^~~~~~~~~~
      |                                                 __ss_padding
/home/peter/.cache/nim/client_d/stdlib_dollars.nim.c:266:50: error: ‘struct sockaddr_storage’ has no member named ‘ss_align’; did you mean ‘__ss_align’?
  266 |  addQuoted__neOpGtcAR9aMi2eKPFrYZHQ((&result), x.ss_align);
      |                                                  ^~~~~~~~
      |                                                  __ss_align
```

This simple PR "fixes" the issue by making sure these fields use the correct name in the C code. I'm not entirely sure why it suddenly tries to create a dollars procedure for this structure, so the issue might lie somewhere else entirely. And if this is indeed the required fix it should probably be applied to more fields/across more files.

For completeness sake, this is the hash procedure I implemented:

``` Nim
proc hash*(sock_addr: Sockaddr_storage): Hash =
  result = cast[array[128, byte]](sock_addr).hash
```

### CODE REVIEW

**Comment by Araq** [comment]
Please add a test case for this so that we can see where else it needs to be fixed.

**Comment by PMunch** [comment]
What kind of test case would that be?

**Comment by PMunch** [comment]
Fixed a similar issue in `winlean`, still no word on what kind of test I should write for this?

File: lib/windows/winlean.nim [hunk 1]
```diff
@@ -517,7 +517,7 @@ type
     ai_family*: cint        ## Address family of socket.
     ai_socktype*: cint      ## Socket type.
     ai_protocol*: cint      ## Protocol of socket.
-    ai_addrlen*: csize_t        ## Length of socket address.
+    ai_addrlen*: csize        ## Length of socket address.
```
Comment (Araq) [inline lib/windows/winlean.nim]: ``csize_t`` is correct.

Comment (PMunch) [reply lib/windows/winlean.nim]: Oh woops, not sure how that snuck in there

File: lib/posix/posix_linux_amd64.nim [hunk 2]
```diff
@@ -433,8 +433,8 @@ type
                        header: "<sys/socket.h>",
                        pure, final.} = object ## struct sockaddr_storage
     ss_family*: TSa_Family ## Address family.
-    ss_padding: array[128 - sizeof(cshort) - sizeof(culong), char]
-    ss_align: clong
+    ss_padding {.importc: "__ss_padding".}: array[128 - sizeof(cshort) - sizeof(culong), char]
+    ss_align {.importc: "__ss_align".}: clong
```
Comment (krux02) [inline lib/posix/posix_linux_amd64.nim:432]: just delete all padding and alignment fields. They are not supposed to be printed as they don't contain any useful data. They are not exported, so code won't fail if they are gone. Size and alignment computation is forwarded to the c compiler for imported types.

File: lib/windows/winlean.nim (hunk 1 above)
Comment (Clyybber) [inline lib/windows/winlean.nim:520]:
```suggestion
    ai_addrlen*: csize_t        ## Length of socket address.
```

Comment (Clyybber) [reply lib/windows/winlean.nim]:
```suggestion
    ai_addrlen*: csize_t        ## Length of socket address.
```

**Comment by PMunch** [comment]
Well shit.. Tried to rebase to keep Stalebot happy but that just created a mess out of the PR..

**Pull request merged**